### PR TITLE
Global field

### PR DESCRIPTION
**Pull request by rohitmishra209**

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ module.exports = {
   // pass locale, only to migrate entries from that locale
   // not passing `locale` will migrate all the locales present
   // locales: ['fr-fr'],
-  host: 'https://api.contentstack.io/v3',
+  host: 'https://api.contentstack.com/v3',
```
Comment (ninadhatkar) [inline config/default.js:6]: Can you verify this?

File: config/index.js [hunk 2]
```diff
@@ -9,5 +9,5 @@ module.exports = {
   password: '',
   target_stack: '',
   // Folder in which exported contents are stored
-  data: './contents'
+  data: './New-data'
```
Comment (ninadhatkar) [inline config/index.js:12]: Can you verify this?

File: lib/import/global_fields.js [hunk 3]
```diff
@@ -74,9 +87,10 @@ importGlobalFields.prototype = {
             log.success(global_field_uid +' '+' globalfield created successfully');
             return;
           }).catch(function (error) {
+            console.log("dattatatat", _globalField_pending);
             if(error.error_code === 115) {
               // eslint-disable-next-line no-console
-              log.error('Globalfield fail to import '+ JSON.stringify(error.errors));  
+              log.error('globalfield fail to import '+ JSON.stringify(error.errors));
```
Comment (ninadhatkar) [inline lib/import/global_fields.js:93]: Suggestion for log 'Global Field: Failed to import due to: ' + JSON.stringify(error.errors)

File: lib/util/lookupReplaceAssets.js [hunk 4]
```diff
@@ -26,6 +26,7 @@ module.exports = function (data, mappedAssetUids, mappedAssetUrls, assetUidMappe
   var matchedUrls = [];
 
   var find = function (schema, entry) {
+   // console.log("findingggg", schema);
```
Comment (ninadhatkar) [inline lib/util/lookupReplaceAssets.js:29]: remove  unused comments

File: lib/import/global_fields.js (hunk 3 above)
Comment (rohitmishra209) [reply lib/import/global_fields.js:93]: Have done because all other messages are in small cases the error log.

File: config/default.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
 module.exports = {
-  versioning: true,
+  versioning: false,
   // pass locale, only to migrate entries from that locale
   // not passing `locale` will migrate all the locales present
   // locales: ['fr-fr'],
-  host: 'https://api.contentstack.io/v3',
+  host: 'https://api.contentstack.com/v3',
```
Comment (ninadhatkar) [inline config/default.js:6]: Verify this

File: lib/util/lookupReplaceAssets.js (hunk 4 above)
Comment (rohitmishra209) [reply lib/util/lookupReplaceAssets.js]: done

File: lib/import/content_types.js [hunk 6]
```diff
@@ -142,45 +143,37 @@ importContentTypes.prototype = {
...
               helper.writeFile(globalFieldUpdateFile, globalFieldMapperFolderpath);
               return;
             }).catch(function (error) {
-              // console.log("dattatatat", _globalField_pending);
-              if(error.error_code === 115) {
-                // eslint-disable-next-line no-console
-                log.error('globalfield fail to import '+ JSON.stringify(error.errors));  
-              }
+              // eslint-disable-next-line no-console
+              log.error('Globalfield fail to import '+ JSON.stringify(error.errors));
```
[the remark this answers is not in the document]
Comment (rohitmishra209) [reply lib/import/content_types.js]: done

File: lib/import/content_types.js [hunk 7]
```diff
@@ -17,9 +17,10 @@ var log = require('../util/log');
 var supress = require('../util/extensionsUidReplace');
 var config = util.getConfig();
 
+var concurrencyNo = config.concurrency;
```
[the remark this answers is not in the document]
Comment (rohitmishra209) [reply lib/import/content_types.js]: kept reqConcurrency

File: lib/import/environments.js [hunk 8]
```diff
@@ -68,9 +68,11 @@ importEnvironments.prototype = {
             helper.writeFile(envUidMapperPath, self.envUidMapper);
             return;
           }).catch(function (error) {
-            self.fails.push(env);
-            // eslint-disable-next-line no-console
-            log.error('Environment: \'' + env.name + '\' failed to be imported\n ' + JSON.stringify(error.errors));
+            if(error.errors.name[0]) {
+              log.success('Environment: \'' + env.name + '\' already exists');
+            } else {
+              log.error('Environment: \'' + env.name + '\' failed to be imported\n ' + JSON.stringify(error.errors));
```
[the remark this answers is not in the document]
Comment (rohitmishra209) [reply lib/import/environments.js]: done

**Comment by rohitmishra209** [comment]
Now you can use Global fields as blocks of a Modular Blocks field and add them within the Group field. You can also add Modular Blocks, Group, and Reference fields to the Global field schema

**Pull request merged**